src/ui/widgets.py

```diff
@@ -79,14 +99,90 @@ def __init__(self):
         self.setCentralWidget(centralWidget)
         # Set the position and size of the main window
         self.positionInTopRightCorner()
-        # Lock the width and height of the window
-        self.setFixedSize(MainWidget.width, MainWidget.height)
+
+    def setupCamera(self):
+        self.capture = cv2.VideoCapture(0)
```

Comment:
This line of code will crash the program if the UI thread is running at the same time as the main thread, since there is only one camera [0] and the main thread is already accessing it.

That brings us to our need for inter-thread communication of some sort, so the main thread can send the detected pupil coordinates to the UI thread, without the UI thread having to re-detect them with the already-in-use camera.

Reply:
With the way I have it currently set up, it won't crash the program. I've written the UI in such a way that it accesses the camera and feeds that data to your code for eye detection. This was necessary since we want the user to be able to see their face and the eye detection in our main window.

Reply:
I do agree that the IPC will be necessary to pass the eye coords so we can perform blink detection + mouse movement, but was planning on saving this for the next PR since this one is already large.